src/Raven.Server/Documents/Queries/CollectionQueryEnumerable.cs

```diff
@@ -306,15 +347,44 @@ private long Initialize()
                     {
                         count++;
 
-                        var doc = _fieldsToFetch.IsProjection
-                            ? _resultsRetriever.GetProjectionFromDocument(document, null, QueryResultRetrieverBase.ZeroScore, _fieldsToFetch, _context, null)
-                            : _inner.Current;
+                        Document doc;
+
+                        if (_fieldsToFetch.IsProjection)
+                        {
+                            var result = _resultsRetriever.GetProjectionFromDocument(document, null, QueryResultRetrieverBase.ZeroScore, _fieldsToFetch, _context, null);
+                            if (result.Document != null)
+                            {
+                                if (IsStartingPoint(result.Document))
+                                    break;
+                            }
+                            else if (result.List != null)
+                            {
+                                bool match = false;
+                                foreach (Document item in result.List)
+                                {
+                                    if (IsStartingPoint(item))
+                                    {
+                                        match = true;
+                                        break;
+                                    }
+                                }
 
-                        if (doc.Data.Count <= 0)
-                            continue;
```

Comment:
Previously we did `continue`. In current code I don't see it anymore - is that okay?

Reply:
Was moved to `IsStartingPoint`

Reply:
`doc.Data.Count <= 0` was moved to `IsStartingPoint` but my question was that now we never have `continue` in the code. 

In old code we had `continue` so we iterated to next document in the loop:

`foreach (var document in _documents.GetDocumentsFrom(_context, _collection, 0, start, _query.PageSize))`

now we have only `break` so I believe we might break the loop too early

Reply:
No, this is fine. Previously, we would continue if `Count == 0`, now we break if `Count > 0`.
Otherwise, we just go back to the loop again. The logic was reversed.